### PR TITLE
feat(console): sign in experience preview

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add iframe preview in sign in experience, the url in iframe is now in draft, will align with UI later.

## Linear Issue Reference

LOG-2388

## Testing

截屏2022-05-10 上午11 30 30

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/components/Preview.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+type Props = {
+  signInExperience: SignInExperience;
+};
+
+const Preview = ({ signInExperience }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const [language, setLanguage] = useState<Language>(Language.English);
+  const [mode, setMode] = useState<AppearanceMode>(AppearanceMode.LightMode);
+  const [platform, setPlatform] = useState<string>('web');
```
Comment (xiaoyijun) [inline packages/console/src/pages/SignInExperience/components/Preview.tsx:20]:
```suggestion
  const [platform, setPlatform] = useState<'web' | 'mobile'>('web');
```

**Review by xiaoyijun — APPROVED** [review]
Overall LGTM.

File: packages/console/src/pages/SignInExperience/components/Preview.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          <Select
+            value={language}
+            options={[
+              { value: Language.English, title: t('sign_in_exp.preview.languages.english') },
+              { value: Language.Chinese, title: t('sign_in_exp.preview.languages.chinese') },
+            ]}
+            onChange={(value) => {
+              setLanguage(value as Language);
+            }}
+          />
```
Comment (demonzoo) [inline packages/console/src/pages/SignInExperience/components/Preview.tsx:44]: nit: Select component should better be able to infer the value type

File: packages/console/src/pages/SignInExperience/components/Preview.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          isActive={platform === 'mobile'}
+          onClick={() => {
+            setPlatform('mobile');
+          }}
+        >
+          {t('sign_in_exp.preview.mobile')}
+        </TabNavItem>
+      </TabNav>
+      <div className={styles.body}>
+        <iframe src={`/sign-in?config=${config}`} />
```
Comment (demonzoo) [inline packages/console/src/pages/SignInExperience/components/Preview.tsx:76]: Does `sign-in` already accept these configs?

Comment (demonzoo) [reply packages/console/src/pages/SignInExperience/components/Preview.tsx:76]: Would it be nice if we send these configs using `postMessage`?

Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/Preview.tsx:76]: Yeah, but we havn't decide how to communicate. Will do it later.

File: packages/console/src/pages/SignInExperience/components/Preview.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/components/Preview.tsx:44]: created LOG-2415, since is not necessary in this PR.

**Review by demonzoo — APPROVED** [review]

File: packages/console/src/pages/SignInExperience/components/Preview.tsx (hunk 2 above)
Comment (demonzoo) [reply packages/console/src/pages/SignInExperience/components/Preview.tsx:46]: Done. #1071

**Pull request merged**